src/NetMQ/ReceiveThreadSafeSocketExtensions.cs

```diff
@@ -18,14 +19,24 @@ public static class ReceiveThreadSafeSocketExtensions
         /// Receive a bytes from <paramref name="socket"/>, blocking until one arrives.
         /// </summary>
         /// <param name="socket">The socket to receive from.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
         /// <returns>The content of the received message.</returns>
+        /// <exception cref="System.OperationCanceledException">The token has had cancellation requested.</exception>
         [NotNull]
-        public static byte[] ReceiveBytes([NotNull] this IThreadSafeInSocket socket)
+        public static byte[] ReceiveBytes([NotNull] this IThreadSafeInSocket socket,
+            CancellationToken cancellationToken = default)
         {
             var msg = new Msg();
             msg.InitEmpty();
 
-            socket.Receive(ref msg);
+            try
+            {
+                socket.Receive(ref msg, cancellationToken);
+            }
+            finally
+            {
+                msg.Close();
```

Comment:
Is it safe to close it then use it below?

Reply:
good catch